pkg/controller/nodes/node_exec_context.go

```diff
@@ -289,19 +289,21 @@ func (c *nodeExecutor) BuildNodeExecutionContext(ctx context.Context, executionC
 	if config.GetConfig().NodeConfig.IgnoreRetryCause {
 		// For the unified retry behavior we execute the last interruptibleFailureThreshold attempts on a non
 		// interruptible machine
-		currentAttempt := s.GetAttempts() + 1 + s.GetSystemFailures()
-		maxAttempts := uint32(config.GetConfig().NodeConfig.DefaultMaxAttempts)
+		currentAttempt := int32(s.GetAttempts() + s.GetSystemFailures())
+		maxAttempts := int32(config.GetConfig().NodeConfig.DefaultMaxAttempts)
 		if n.GetRetryStrategy() != nil && n.GetRetryStrategy().MinAttempts != nil && *n.GetRetryStrategy().MinAttempts != 0 {
-			maxAttempts = uint32(*n.GetRetryStrategy().MinAttempts)
+			maxAttempts = int32(*n.GetRetryStrategy().MinAttempts)
 		}
 
-		if interruptible && currentAttempt >= maxAttempts-c.interruptibleFailureThreshold {
+		if interruptible && ((c.interruptibleFailureThreshold > 0 && currentAttempt >= c.interruptibleFailureThreshold) ||
```

Comment:
If we understand correctly, one can specify a positive and a negative number for both the old and the new behaviour. Is there a place in flytepropeller where the config is sanity checked and rules can be enforced early on?
If yes, what do you think about simply requiring a negative number in the config if the new behaviour is chosen and a positive one in the other case?
This would make the logic here simpler.

Also, what do you think of moving the checks whether the retry is interruptible into two separate functions with descriptive names? This would make the if conditions here easier to understand. Also would make this specific logic easier to unit test.